docs/guide/callbacks.rst

```diff
@@ -0,0 +1,328 @@
+.. _callbacks:
+
+Callbacks
+=========
+
+A callback is a set of functions that will be called at given stages of the training procedure.
+You can use callbacks to access internal state of the RL model during training.
+It allows one to do monitoring, auto saving, model manipulation, progress bars, ...
+
+
+Custom Callback
+---------------
+
+To build a custom callback, you need to create a class that derives from `BaseCallback`.
+This will give you access to events (`_on_training_start`, `_on_step`) and useful variables (like `self.model` for the RL model).
```

Comment:
On the emphasis of code-references: This would be closer to other docs if all code-related text was `formatted like this`, but on the other hand the documentation would be pain to look at with constantly changing styles.

Reply:
I'm not sure what you mean there...

Reply:
Now words like "_on_training_start" are emphasized with italics *like this*, while in most of the documentation all names referring to methods and such are done in preformatted style `like this`. I am not sure which one would be better here, though, as things could get messy with `this style` being all around.

Reply:
My bad... I used markdown syntax  \`some_code\` instead of rst syntax  \`\`some_code\`\`, I will change that.

Reply:
I did some regex magic...most of the code snippets are unified now (I hope I did not break anything).